space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+# Space Framework
+
+## Terminology 
+**Physical Space**, **pSpace** An entity that behaves like a Kubernetes kube-apiserver (including the persistent storage behind it) and the subset of controllers in the kube-controller-manager that are concerned with API machinery generalities (without the management of containerized workloads). A kcp logical cluster is an example. A regular Kubernetes cluster is another example.
+
+**Space Object**, **Space** - A kind of Kubernetes API object that _represents_ a pSpace.  Its "kind" is `Space`, we sometimes include the "object" part when writing specifically about the API objects to clearly distinguish them from what they represent.
+
+**Space Provider** - A Thing that manages the life cycle of multiple pSpaces. This includes at minimal create/delete/List operations on pSpaces. KIND is an example for Space Provider (that manages KIND clusters), KCP is another example - KCP manages KCP's workspaces. 
+
+## High level architecture  & Main components
+![SF overall view](SF-All.drawio.svg "SF overall view")
+
+The Space Framework (SF) is a generic management framework for space providers and pSpaces. The framework defines an abstraction layer for space providers and pSpace management that allows clients (both CLI/script based and client-go based clients) to use spaces while maintaining the clients decoupled from the specific pSpace and space provider that is being used. 
```

Comment:
There is an inconsistency here. In the Terminology section it is said that "KIND is an example for Space Provider". The sentence on which I am commenting here does not speak specifically of SpaceProviderDesc and SpaceProviderType, it only speaks of "space providers". If such a "space provider" is a SpaceProviderDesc then KIND is not an example of that (KIND is an example of SpaceProviderType). If "space provider" here means SpaceProviderType then omission of SpaceProviderDesc is leaving out a critical part.

Reply:
I'm not sure I understand the issue here. I don't see any reference to KIND here. Maybe it's outdated text ?

Reply:
I still see "KIND is an example for Space Provider" in the Terminology section entry for "Space Provider".